app/code/Magento/CmsGraphQl/Model/Resolver/DataProvider/Page.php

```diff
@@ -69,4 +90,56 @@ public function getData(int $pageId): array
         ];
         return $pageData;
     }
+
+    /**
+     * @param int $pageId
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    public function getDataByPageId(int $pageId): array
+    {
+        $page = $this->pageRepository->getById($pageId);
+
+        return $this->convertPageData($page);
+    }
+
+    /**
+     * @param string $pageIdentifier
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    public function getDataByPageIdentifier(string $pageIdentifier): array
+    {
+        $storeId = (int)$this->storeManager->getStore()->getId();
+        $page = $this->pageByIdentifier->execute($pageIdentifier, $storeId);
+
+        return $this->convertPageData($page);
+    }
+
+    /**
+     * @param PageInterface $page
+     * @return array
+     * @throws NoSuchEntityException
+     */
+    private function convertPageData(PageInterface $page)
```

Comment:
3. Reuse the newly created method `convertPageData` in `getData` method - so there will be no code duplication;

Reply:
**done**

Could you please review all changes now?

Thank you!